src/main/java/seedu/address/model/TaskList.java

```diff
@@ -61,6 +61,18 @@ public void addTask(Task t) {
         tasks.add(t);
     }
 
+    /**
+     * Check a task for duplicates
+     */
+    public boolean hasTask(Task t) {
+        for (int i = 0; i < tasks.size(); i++) {
```

Comment:
You can use the predefined "contains" function from ObservableList

Reply:
I have refactored the method to shorten code using contains